### PR TITLE
[ST-2] Try all operations on simple examples (create, update, select, delete)

### PR DESCRIPTION
**Pull request by netelli**

### CODE REVIEW

File: src/main/java/com/github/netelli/JDBCStuding.java [hunk 1]
```diff
@@ -1,45 +1,28 @@
...
-            insertData(statement);
+        ProductsDAO productsDAO = new ProductsDAO("jdbc:h2:mem:test");
 
-            ResultSet rs = statement.executeQuery("select * from products");
-            while (rs.next()) {
-                System.out.println(rs.getInt("id") + " : " + rs.getString("title") + " : " + rs.getInt("categoryId"));
-            }
-        }
-    }
+        productsDAO.getConnection();
```
Comment (fakeone) [inline src/main/java/com/github/netelli/JDBCStuding.java]: better rename this method to init, or set up connection, there is no need to return this connection from class and break encapsulation.

Comment (netelli) [inline src/main/java/com/github/netelli/JDBCStuding.java]: Done

File: src/main/java/com/github/netelli/ProductsDAO.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    public ProductsDAO(String jdbcUrl) {
+        this.jdbcUrl = jdbcUrl;
+    }
+
+    public void deleteData() throws SQLException {
+        Statement statement = connection.createStatement();
+        statement.execute("delete from products where brandId = 2");
+    }
+
+    public ResultSet getData(String tableName) throws SQLException {
```
Comment (netelli) [inline src/main/java/com/github/netelli/ProductsDAO.java]: Done

File: src/main/java/com/github/netelli/JDBCStuding.java [hunk 3]
```diff
@@ -1,45 +1,28 @@
...
+        String tableName = "products";
+        ResultSet rs = productsDAO.getData(tableName);
+        productsDAO.displayData(rs);
 
-    private static void createTables(Statement statement) throws SQLException {
-        statement.execute("create table categories(" +
-                "id integer primary key auto_increment, " +
-                "title varchar(100));");
+        productsDAO.updateData();
+        productsDAO.displayData(productsDAO.getData(tableName));
```
Comment (netelli) [inline src/main/java/com/github/netelli/JDBCStuding.java]: Done

File: src/main/java/com/github/netelli/JDBCStuding.java [hunk 4]
```diff
@@ -1,45 +1,28 @@
...
-                "title varchar(100));");
+        productsDAO.deleteData();
+        productsDAO.displayData(productsDAO.getData(tableName));
 
-        statement.execute("create table products(" +
-                "id integer primary key auto_increment, " +
-                "title varchar(100), categoryId integer, brandId integer" +
-                "constraint categoryId foreign key (id) references categories (id)" +
-                "constraint brandId foreign key (id) references brands (id));");
+        productsDAO.closeConnection();
```
Comment (netelli) [inline src/main/java/com/github/netelli/JDBCStuding.java]: Done

File: src/main/java/com/github/netelli/ProductsDAO.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+import java.sql.*;
+
+/**
+ * Created by user on 08.12.2016.
+ */
+public class ProductsDAO {
+
+    private final String jdbcUrl;
+    Connection connection = null;
```
Comment (netelli) [inline src/main/java/com/github/netelli/ProductsDAO.java]: Done

File: src/main/java/com/github/netelli/ProductsDAO.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        return connection;
+    }
+
+    public void closeConnection() {
+        try {
+            if (connection != null) {
+                connection.close();
+            }
+        } catch (Exception e) {
+            //do nothing
```
Comment (netelli) [inline src/main/java/com/github/netelli/ProductsDAO.java]: Done

File: src/main/java/com/github/netelli/ProductsDAO.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    public void displayData(ResultSet rs) throws SQLException {
+        System.out.println(("ID") + " | " + ("TITLE") + " | " + ("CATEGORY") + " | " + ("BRAND"));
+        while (rs.next()) {
+            System.out.println(rs.getInt("id") + " : " + rs.getString("title") + " : " + rs.getInt("categoryId") + " : " + rs.getInt("brandId"));
+        }
+        System.out.println("_______________________________");
+    }
+
+    public void updateData() throws SQLException {
+        Statement statement = connection.createStatement();
```
Comment (netelli) [inline src/main/java/com/github/netelli/ProductsDAO.java]: Done

File: src/main/java/com/github/netelli/ProductsDAO.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        statement.execute("update products set brandId = 2 where id = 2");
+    }
+
+    public void insertData() throws SQLException {
+        Statement statement = connection.createStatement();
```
Comment (netelli) [inline src/main/java/com/github/netelli/ProductsDAO.java]: Ok

File: test/com/github/netelli/ProductsDAOTest.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    public void testInsertData() throws Exception {
+        ProductsDAO productsDAO = new ProductsDAO("jdbc:h2:mem:test");
+        productsDAO.getConnection();
+        productsDAO.createTables();
+        productsDAO.insertData();
+
+        ResultSet rs = productsDAO.getData("products");
+
+        int count = 0;
+        ArrayList<String> products = new ArrayList<>();
```
Comment (netelli) [inline test/com/github/netelli/ProductsDAOTest.java]: Done

File: test/com/github/netelli/ProductsDAOTest.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+        while (rs.next()) {
+            count++;
+            products.add(rs.getString("title"));
+        }
+
+        assertTrue(count == 3);
```
Comment (netelli) [inline test/com/github/netelli/ProductsDAOTest.java]: Done

File: test/com/github/netelli/ProductsDAOTest.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+        assertTrue("skirt mini".equals(products.get(0)));
```
Comment (netelli) [inline test/com/github/netelli/ProductsDAOTest.java]: Done

File: test/com/github/netelli/ProductsDAOTest.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+        productsDAO.closeConnection();
```
Comment (netelli) [inline test/com/github/netelli/ProductsDAOTest.java]: Done

**Comment by netelli** [comment]
Batja, done

**Pull request merged**